### PR TITLE
openldap: update to 2.6.6

### PR DESCRIPTION
**Pull request by klarasm**
The 2.4 branch is not maintained anymore and I don't know if there's much advantage to going to 2.5 instead of 2.6

Some shared libraries have changed name so this updates common/shlibs.

As this is a major update the ldap database should be backed up and recreated as per https://www.openldap.org/doc/admin26/maintenance.html

#### Testing the changes
- I tested the changes in this PR: **YES** (mainly openldap, mit-krb5 and cyrus-sasl)
- I haven't tested the revbumped packages that much, other than building with their test suite

I have upgraded my ldap setup to this version. It's not very complicated so may not be great for coverage.
Basically:
- two nodes, with normal syncrepl and gssapi/kerberos for authentication
- the ldap directory is used as database backend for my kerberos setup
- I have a few machines setup with kerberos for authentication and ldap as id provider (some with sssd and some with nslcd)

This PR should not be merged before:
- [x] log4cxx-1.0.0 #42632
- [x] mit-krb5 1.20.1 #42633
- [x] python3-ldap-3.4.3 #42634
- [x] apr-util-1.6.3 #42636 (change in libtool resulted in cross-compilation failure)
- [x] virtuoso-7.2.9 #42666 (probably not critical, tests are already failing on master branch)
- [x] apache-2.4.56 #43108 (same failure as in apr-util)

[ci skip] again.

#### Local build testing
(updated 2023-08-23)

##### I built this PR locally for my native architecture, (x86_64-glibc):
- Build failures:
  - alpine (already failing on master branch)
- Test suite errors:
  - anjuta  (already failing on master branch)

##### I built this PR locally for my native architecture, (x86_64-musl):
- Build failures:
  -  sssd (does not support musl)
  -  alpine (already failing on master branch)
- Test suite errors:
  - anjuta (already failing on master branch)
  - audit (already failing on master branch)
  - dovecot (already failing on master branch)
  - nss-pam-ldapd (musl does not support nsswitch)

##### Crossbuilds
- aarch64-musl, build failures:
  - alpine (already failing on master branch)
  - sssd (does not support musl)
  - packages not supporting cross-compilation:
    - FreeRADIUS
    - libreoffice
    - virtuoso
    - x2goclient
- armv7l, build failures:
  - alpine (already failing on master branch)
  - packages not supporting cross-compilation:
    - FreeRADIUS
    - libreoffice
    - virtuoso
    - x2goclient
- armv6l-musl, build-failures:
  - alpine (already failing on master branch)
  - sssd (does not support musl)
  - packages not supporting cross-compilation:
    - FreeRADIUS
    - libreoffice
    - virtuoso
    - x2goclient

### CODE REVIEW

**Comment by klarasm** [comment]
Turns out test079 fails because arguments after the filter for ldapsearch seem to be ignored on musl. Added a patch that moves the filter last.

**Comment by klarasm** [comment]
The test suite for mit-krb5 seems to fail now instead. That package is out of date as well. May try to update it and see if it solves the problem.

**Comment by klarasm** [comment]
Still lacking a few revbumps. Will look into them later.

**Comment by klarasm** [comment]
That should be most revbumps. Now to see about build testing.

**Comment by klarasm** [comment]
Fix some build issues. Still have failures in anjuta, log4cxx and virtuoso.

**Comment by klarasm** [comment]
Fix log4cxx build issue.

**Comment by klarasm** [comment]
Fixed anjuta build issue.

**Comment by klarasm** [comment]
Managed to get newest version of virtuoso to build but it still fails the test suite.

**Comment by klarasm** [comment]
The test suite for vitruoso was failing due to missing checkdepends...
That was probably the cause on the previous version as well.

All revbumped or updated packages now complete their test suite if they have one on x86_64-glibc. I'm going to run this on x86_64-musl as well but it will take longer as that machine is a lot less powerful.

**Comment by klarasm** [comment]
On x86_64-musl tests fail for audit, dovecot-plugin-pigeonhole, dovecot, evolution-data-server, evolution and nss-pam-ldapd.
Additionally sssd fails to build.

nss-pam-ldapd and sssd are expected as musl lacks support for nsswitch.

On dovecot and dovecot-plugin-pigeonhole a single test fails:
```
test-common.c:245: Assert failed: suppress == TRUE
Error: mmap_anon(.test_file_cache, 4096) failed: Out of memory
test-common.c:245: Assert failed: suppress == TRUE
Error: mremap_anon(.test_file_cache, 8192) failed: Out of memory
file_cache_errors .................................................... : FAILED
```

audit fails in auparse test:
```
make[4]: Entering directory '/builddir/audit-3.0.3/auparse/test'
test "../.." = "../.." || \
                cp ../../auparse/test/test*.log .
LC_ALL=C \
./auparse_test > auparse_test.cur
Config file /etc/audit/auditd.conf doesn't exist, skipping
make[4]: *** [Makefile:709: check-local] Error 1
make[4]: Leaving directory '/builddir/audit-3.0.3/auparse/test'
make[3]: *** [Makefile:580: check-am] Error 2
make[3]: Leaving directory '/builddir/audit-3.0.3/auparse/test'
make[2]: *** [Makefile:2040: check-recursive] Error 1
make[2]: Leaving directory '/builddir/audit-3.0.3/auparse'
make[1]: *** [Makefile:2192: check] Error 2
make[1]: Leaving directory '/builddir/audit-3.0.3/auparse'
make: *** [Makefile:467: check-recursive] Error 1
=> ERROR: audit-3.0.3_5: do_check: '${make_check_pre} ${make_cmd} ${makejobs} ${make_check_args} ${make_check_target}' exited with 2
=> ERROR:   in do_check() at common/build-style/gnu-configure.sh:33
```

For evolution and evolution-data-server a bunch of tests fail with similar output to:
```
      Start  2: test-ebook-get-contact                                                                                                                         
 2/90 Test  #2: test-ebook-get-contact ....................SIGTRAP***Exception:   2.58 sec                                                                     
# random seed: R02Sd090595520940fe58d9ab8434502e157                                                                                                            
1..2                                                                                                                                                           
# Start of EBook tests                                                                                                                                         
# Start of GetContact tests                                                                                                                                    
# GLib-GIO-DEBUG: _g_io_module_get_default: Found default implementation memory (GMemorySettingsBackend) for *gsettings-backend*                               
# GLib-GIO-DEBUG: Using cross-namespace EXTERNAL authentication (this will deadlock if server is GDBus < 2.73.3)                                               
dbus-daemon[30863]: [session uid=1000 pid=30863] Activating service name='org.gnome.evolution.dataserver.Sources5' requested by ':1.0' (uid=1000 pid=30861 comm
="/builddir/evolution-data-server-3.46.3/build/tests")                                                                                                         
dbus-daemon[30863]: [session uid=1000 pid=30863] Successfully activated service 'org.gnome.evolution.dataserver.Sources5'                                      
dbus-daemon[30863]: [session uid=1000 pid=30863] Activating service name='org.gnome.evolution.dataserver.AddressBook10' requested by ':1.0' (uid=1000 pid=30861
 comm="/builddir/evolution-data-server-3.46.3/build/tests")                                                                                                    
dbus-daemon[30863]: [session uid=1000 pid=30863] Successfully activated service 'org.gnome.evolution.dataserver.AddressBook10'                                 
Bail out! e-test-server-utils-FATAL-ERROR: Unable to create the test book: Message recipient disconnected from message bus without replying                    
                                                                                                                                                               
(/builddir/evolution-data-server-3.46.3/build/tests/libebook/test-ebook-get-contact:30861): e-test-server-utils-ERROR **: 10:03:46.504: Unable to create the te
st book: Message recipient disconnected from message bus without replying                                                                                      
cleaning up pid 30863                                                                                                                                          
```

[audit.log](https://github.com/void-linux/void-packages/files/10568583/audit.log)
[dovecot.log](https://github.com/void-linux/void-packages/files/10568549/dovecot.log)
[dovecot-plugin-pigeonhole.log](https://github.com/void-linux/void-packages/files/10568550/dovecot-plugin-pigeonhole.log)
[evolution.log](https://github.com/void-linux/void-packages/files/10568551/evolution.log)
[evolution-data-server.log](https://github.com/void-linux/void-packages/files/10568552/evolution-data-server.log)

**Comment by klarasm** [comment]
Did some tests on the master branch and x86_64-musl.
audit, dovecot and evloution-data-server already fails so those may not be related to this PR.
evolution did not find any test for some reason and thus passed.

When I looked closer on dovecot-plugin-pigeonhole it fails in this PR because it couldn't build dovecot as the test suite is failing. If I build dovecot first without checks dovecot-plugin-pigeonhole it succeeds with the commits in this PR.

Updated log for dovecot pigeonhole in this PR:
[dovecot-plugin-pigeonhole.log](https://github.com/void-linux/void-packages/files/10569418/dovecot-plugin-pigeonhole.log)

Logs  from builds on master branch:
[audit.log](https://github.com/void-linux/void-packages/files/10569284/audit.log)
[dovecot.log](https://github.com/void-linux/void-packages/files/10569285/dovecot.log)
[dovecot-plugin-pigeonhole.log](https://github.com/void-linux/void-packages/files/10569286/dovecot-plugin-pigeonhole.log)
[evolution.log](https://github.com/void-linux/void-packages/files/10569287/evolution.log)
[evolution-data-server.log](https://github.com/void-linux/void-packages/files/10569288/evolution-data-server.log)

So from what I can see there are no new failures caused by this PR. I'll do some more crossbuilds but I think after that this PR is starting to be ready.

Regarding the backup/restore of the ldap database, should that be notified on upgrade? Skipping this step can potentially corrupt the database but I guess anyone using openldap in server mode should be aware of that and have periodic backups anyway but I don't know if that's something we can depend on.

**Comment by klarasm** [comment]
I feel confident enough about the changes that I will mark this as ready. Please tell me if there's any other testing or changes that should be done.

**Comment by dkwo** [comment]
Do you mind checking that these build with openssl 3?
If not, I can do it, but I think `squid` may need update to 5.7 to work.
#37681

**Comment by dkwo** [comment]
See https://github.com/void-linux/void-packages/pull/42054
I'll close it if you prefer to do it here.

**Comment by klarasm** [comment]
I'll give openssl 3 a go. I don't mind you updating squid in the separate PR, but will pull in that as well in my testing. This will probably take longer to merge than your PR.

**Comment by kruceter** [comment]
sssd is already faulty as is.

I was trying to update it in https://github.com/void-linux/void-packages/pull/40846, but I have no means to test it since I have nothing to set it up for.

**Comment by klarasm** [comment]
> sssd is already faulty as is.
> 
> I was trying to update it in #40846, but I have no means to test it since I have nothing to set it up for.

sssd will probably not work on musl as musl lacks native support for nsswitch. There is musl-nscd that implements this via nscd but I haven't been able to get that to work when I tried it earlier.
Maybe sssd could be patched so only the pam portion gets built but that would limit the functionality of it. At that point it's probably better to use them pam portion of nss-pam-ldapd or pam-krb5.

I could spin up a void linux machine with glibc to test sssd though.

**Comment by kruceter** [comment]
> sssd will probably not work on musl

I tested it briefly and it works on musl, plus alpine has it (but with disabled tests).

**Comment by klarasm** [comment]
> > sssd will probably not work on musl
> 
> I tested it briefly and it works on musl, plus alpine has it (but with disabled tests).

Okay, that would be great! I'll do some testing on your branch in that case

File: srcpkgs/alpine/template [hunk 1]
```diff
@@ -29,7 +29,3 @@ do_build() {
 		CC="$BUILD_CC" CFLAGS="$BUILD_CFLAGS" CPPFLAGS= LDFLAGS="$BUILD_LDFLAGS"
 	make ${makejobs} all CC="$CC"
 }
-
-post_install() {
-	vlicense NOTICE
-}
```
Comment (paper42) [inline srcpkgs/alpine/template:35]: why are you removing this?

File: srcpkgs/log4cxx/template [hunk 2]
```diff
@@ -1,22 +1,21 @@
...
 pkgname=log4cxx
-version=0.10.0
-revision=4
-build_style=gnu-configure
-makedepends="libxml2-devel apr-devel apr-util-devel gdbm-devel"
-short_desc="A C++ Logging Library based on Log4J"
+version=1.0.0
+revision=1
+build_style=cmake
+makedepends="libxml2-devel apr-devel apr-util-devel gdbm-devel pkg-config boost-devel zip"
```
Comment (paper42) [inline srcpkgs/log4cxx/template:6]: zip and pkg-config should probably be in hostmakedepends

File: srcpkgs/mit-krb5/template [hunk 3]
```diff
@@ -12,7 +12,7 @@ maintainer="Orphaned <orphan@voidlinux.org>"
 license="MIT"
 homepage="http://web.mit.edu/kerberos"
 distfiles="${homepage}/dist/krb5/${version%.*}/krb5-${version}.tar.gz"
```
Comment (paper42) [inline srcpkgs/mit-krb5/template:14]:
```suggestion distfiles="http://web.mit.edu/kerberos/dist/krb5/${version%.*}/krb5-${version}.tar.gz"
```

File: srcpkgs/python3-ldap/template [hunk 4]
```diff
@@ -13,7 +13,7 @@ maintainer="Michael Aldridge <maldridge@voidlinux.org>"
 license="Python-2.0"
 homepage="https://www.python-ldap.org"
 distfiles="${PYPI_SITE}/p/${_pkgname}/${_pkgname}-${version}.tar.gz"
-checksum=7d1c4b15375a533564aad3d3deade789221e450052b21ebb9720fb822eccdb8e
+checksum=ab26c519a0ef2a443a2a10391fa3c5cb52d7871323399db949ebfaa9f25ee2a0
 
 pre_build() {
 	sed -i "s:^libs = .*:libs = ldap_r lber sasl2 ssl crypto:" setup.cfg
```
Comment (paper42) [inline srcpkgs/python3-ldap/template:19]: can we switch to vsed here?

File: srcpkgs/seahorse/patches/allow-building-with-gpg-2.4.patch [hunk 5]
```diff
@@ -0,0 +1,13 @@
+diff --git a/meson.build b/meson.build
+index e29b532236e5543682832725d79dea0c7e49f8bf..23d0b54f567c112249fc72ddb91ac8e40218c60b 100644
+--- a/meson.build
++++ b/meson.build
+@@ -26,7 +26,7 @@ endif
+ # Dependencies
+ min_glib_version = '2.66'
+ min_gcr_version = '3.38'
+-accepted_gpg_versions = [ '2.2.0', '2.3.0' ]
++accepted_gpg_versions = [ '2.2.0', '2.3.0', '2.4.0' ]
```
Comment (paper42) [inline srcpkgs/seahorse/patches/allow-building-with-gpg-2.4.patch:10]: Could you open a separate PR for this fix?

File: srcpkgs/alpine/template (hunk 1 above)
Comment (klarasm) [reply srcpkgs/alpine/template:35]: xlint complained that license Apache-2.0 should not be installed. I don't know if there's another solution

File: srcpkgs/seahorse/patches/allow-building-with-gpg-2.4.patch (hunk 5 above)
Comment (klarasm) [reply srcpkgs/seahorse/patches/allow-building-with-gpg-2.4.patch:10]: Will do

File: srcpkgs/alpine/template (hunk 1 above)
Comment (paper42) [reply srcpkgs/alpine/template:35]: NOTICE suggests license should probably be `Apache-2.0, BSD-4-Clause`

File: srcpkgs/seahorse/patches/allow-building-with-gpg-2.4.patch (hunk 5 above)
Comment (klarasm) [reply srcpkgs/seahorse/patches/allow-building-with-gpg-2.4.patch:10]: #42055

File: srcpkgs/alpine/template (hunk 1 above)
Comment (klarasm) [reply srcpkgs/alpine/template:35]: Okay, will try that

**Comment by klarasm** [comment]
@dkwo Finished building this PR with #37681 on glibc, have not run the test suites yet. With #42054 squid builds ok. I get failures in packages FreeRADIUS, php8.0 and php (php8.1 is fine though). FreeRADIUS may be able to be resolved by updating.

[FreeRADIUS.log](https://github.com/void-linux/void-packages/files/10581241/FreeRADIUS.log)
[php.log](https://github.com/void-linux/void-packages/files/10581242/php.log)
[php8.0.log](https://github.com/void-linux/void-packages/files/10581243/php8.0.log)

File: srcpkgs/python3-ldap/template (hunk 4 above)
Comment (klarasm) [reply srcpkgs/python3-ldap/template:19]: I actually don't think that expression changes anything here. openldap 2.6 does not provide `ldap_r` anymore, it's `ldap`. I have tried removing the pre_build and it builds and checks fine still.

**Comment by dkwo** [comment]
Thanks, see #42059 I'm aware of the other two.

**Comment by klarasm** [comment]
Upstream has released 2.6.4. This has an upstream fix for test079 failing on musl (https://git.openldap.org/openldap/openldap/-/commit/a858f892b35d463b5b3324063579bd22504f06e8 https://bugs.openldap.org/show_bug.cgi?id=9931).
Will update the PR once I have rerun a few tests.

**Comment by klarasm** [comment]
Earlier I added `libssp-devel` to the hostmakedepends of openldap because a build failure on musl. Turns out I didn't need that and it was old packages in my hostdir that caused the failure.

**Comment by paper42** [comment]
I applied the anjuta build fix separately in 4dfec0707c7472d401c49cef60b5a8e1beae4d80, thanks

File: srcpkgs/python3-ldap/template [hunk 6]
```diff
@@ -1,10 +1,10 @@
 # Template file for 'python3-ldap'
 pkgname=python3-ldap
 _pkgname=python-ldap
```
Comment (paper42) [inline srcpkgs/python3-ldap/template:3]: let's remove this variable and replace it with it's value in $distfiles

**Review by paper42 — COMMENTED** [review]
Could you put all of the updates to separate PRs? It would make them easier to review and merge and this PR would get much simpler.

File: srcpkgs/python3-ldap/template (hunk 6 above)
Comment (klarasm) [reply srcpkgs/python3-ldap/template:3]: Will do.
Some of the packages have new updates and no longer build or test cleanly so will have to fix that as well.
- apr-util: some cross-compilation issues new in 1.6.3
- virtuoso: test suite fails again in 7.2.9

**Comment by klarasm** [comment]
The biggest holdup now is apr-util, a change in the libtool package broke cross-compilation for it and that was not fixed initially. As this PR revbumps apr-util cross-compilation of it would fail without the update (quite a few packages also depend on it).
The virtuoso update is not as critical (the test suite was already failing before).

Otherwise more testing would probably help. I'll rebase this again and rerun tests when I have time.

**Comment by klarasm** [comment]
For local build testing (x86_64-glibc) , distfiles for apache are not available so it and packages that depend on it fail. Additionally one test of anjuta fails.

**Comment by klarasm** [comment]
Modules were broken while crossbuilding (x86_64-glibc to x86_64-musl at least) on the master branch as well. If anyone has any suggestion for how to make it work, feel free to comment.
Otherwise I'll build the overlays directly into slapd as that works.

**Comment by klarasm** [comment]
2.6.5 has been released. Will redo build/check testing for dependents.

**Comment by klarasm** [comment]
Now that apr-util is updated there should probably be nothing major blocking this PR. I'll redo build testing though I don't think much will change (very few changes from 2.6.5 to 2.6.6).

**Comment by klarasm** [comment]
Seems I somehow dropped evolution-data-server when rebasing, sorry about that.

**Pull request merged**